src/test/scala/strawman/collection/test/Test.scala

```diff
@@ -288,6 +289,37 @@ class StrawmanTest {
     println(xs17.to(List))
   }
 
+  def sortedSets(xs: immutable.SortedSet[Int]): Unit = {
+    val xs1 = xs.map((x: Int) => x.toString)
+    val xs2: immutable.SortedSet[String] = xs1
+  }
+
+  def mapOps(xs: Map[Int, String]): Unit = {
+    val xs1 = xs.map((k, v) => (v, k))
+    val xs2: strawman.collection.Map[String, Int] = xs1
+    val xs3 = xs.map(kv => (kv._2, kv._1))
+    val xs4: strawman.collection.Iterable[(String, Int)] = xs3
+    println(xs1)
+    println(xs2)
+    println(xs3)
+    println(xs4)
+  }
+
+  def sortedMaps(xs: immutable.SortedMap[String, Int]): Unit = {
+    val x1 = xs.get("foo")
+    val x2: Option[Int] = x1
+    val xs1 = xs + ("foo", 1)
+    val xs2: immutable.SortedMap[String, Int] = xs1
+    val xs3 = xs.map(kv => kv._1)
+    val xs4: immutable.Iterable[String] = xs3
+    val xs5 = xs.map((k: String, v: Int) => (v, k))
+    val xs6: immutable.SortedMap[Int, String] = xs5
+    class Foo
+//    val xs7 = xs.map((k: String, v: Int) => (new Foo, v)) Error: No implicit Ordering defined for Foo
+    val xs7 = (xs: immutable.Map[String, Int]).map((k, v) => (new Foo, v))
+    val xs8: immutable.Map[Foo, Int] = xs7
```

Comment:
When type signatures are complex, explicit upcasting is a large burden.

Another solution is to add an `asSuper` method that does nothing but the upcast.  (E.g. for `TreeMap` it would be `asMap`.)